src/query/string_opfunc.c

```diff
@@ -581,7 +581,7 @@ db_string_unique_prefix (const DB_VALUE * db_string1, const DB_VALUE * db_string
 
       intl_pad_char (codeset, pad, &pad_size);
 
-      if (collation_id < LANG_COLL_ISO_BINARY_TS)
+      if (collation_id >= LANG_COLL_ISO_BINARY_TI && collation_id < LANG_MAX_BUILTIN_COLLATIONS)
```

Comment:
Is this condition valid even if collation is added by user?

Reply:
this checks the key's collation is ti-type of built-in collation or not.